java/src/main/java/com/google/appengine/tools/cloudstorage/dev/LocalRawGcsService.java

```diff
@@ -593,9 +630,27 @@ public ListItemBatch list(String bucket, String prefix, String delimiter,
     Set<String> prefixes = new HashSet<>();
     QueryResultIterator<Entity> dsResults =
         datastore.prepare(query).asQueryResultIterator(fetchOptions);
+    boolean fromBlobstore = false;
+    if (!dsResults.hasNext()) {
+      // If no results, perhaps there's metadata stored by the Blobstore API that we missed.
+      // Note that Blobstore metadata results aren't returned if some blobs are uploaded via
+      // Blobstore and others are uploaded using this client library.  We only check for Blobstore
+      // metadata if no results are found in appengine-gcs-client's metadata store.
```

Comment:
The reason for this was to avoid extra complexity with storing state information for query continuation.  Since a cursor can either correspond to a appengine-gcs-client metadata query, Blobstore query, or null (but we exhausted appengine-gcs-client's metadata, and now have to move on to Blobstore's metadata).  I was a little hesitant to add this state information because this class isn't necessarily a singleton (as noted in the class javadoc), and I didn't want to modify list()'s method signature since it's a public method that implements an interface for both local and production service.  Let me know your opinion and/or whether you have an alternative solution.